File: 724-FindPivotIndex/724-FindPivotIndex.py
# Last updated: 5/29/2026, 11:54:37 AM
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        t=sum(nums)
        l=0
        for i in range(len(nums)):
            t-=nums[i]
            if l==t:
                return i
            l+=nums[i]
        return -1

File: 724-FindPivotIndex/724-FindPivotIndex.java
// Last updated: 5/29/2026, 11:54:43 AM
class Solution {
    public int pivotIndex(int[] nums) {
        int leftsum=0,totalsum=0;
        for(int i=0;i<nums.length;i++){
            totalsum+=nums[i];
        }
        for(int i=0;i<nums.length;i++){
            totalsum-=nums[i];
            if(leftsum==totalsum){
                return i;
            }
            else{
                leftsum+=nums[i];
            }
        }
        return -1;
    }
}